common_defs.h

```diff
@@ -67,6 +67,28 @@ typedef int64_t s64;
 #  endif
 #endif
 
+#if defined(_MSC_VER)
```

Comment:
`#if defined(_MSC_VER)` => `#ifdef _MSC_VER`.

Also, can you put this this new code in its own "section" above the "Type definitions" one?  Like:

```
/* ========================================================================== */ 
/*                            Target architecture                             */ 
/* ========================================================================== */ 

(new code here)
```

Also since these are defined / undefined style macros and not 1 / 0 style macros, to be safe they should all be `#undef`ined before the correct one is `#define`d.